channels/applicationinsights-channel-js/Tests/Unit/src/StatsBeat.tests.ts

```diff
@@ -0,0 +1,246 @@
+            }
+        });
+
+        this.testCaseAsync({
+            name: "Statsbeat increments success count when fetch sender is called once",
+            useFakeTimers: true,
+            stepDelay: 100,
+            steps: [
+                () => {
+                    this.fetchStub = this.sandbox.stub(window, "fetch").callsFake(() => {
+                        return Promise.resolve(new Response("{}", { status: 200, statusText: "OK" }));
+                    });
+
+                    const config = this.createSenderConfig(TransportType.Fetch);
+                    const { sender } = this.initializeCoreAndSender(config, "000e0000-e000-0000-a000-000000000000");
+
+                    const telemetryItem: ITelemetryItem = {
+                        name: "fake item",
+                        iKey: "testIkey2;ingestionendpoint=testUrl1",
+                        baseType: "some type",
+                        baseData: {}
+                    };
+
+                    this.processTelemetryAndFlush(sender, telemetryItem);
+                }
+            ].concat(PollingAssert.createPollingAssert(() => {
+                if (this.statsbeatCountSpy.called && this.fetchStub.called) {
+                    this.assertStatsbeatCall(200, "Request_Success_Count");
+                    return true;
+                }
+                return false;
+            }, "Waiting for fetch sender and Statsbeat count to be called") as any)
+        });
+
+        this.testCaseAsync({
+            name: "Statsbeat increments retry count when fetch sender is called once",
+            useFakeTimers: true,
+            stepDelay: 100,
+            steps: [
+                () => {
+                    this.fetchStub = this.sandbox.stub(window, "fetch").callsFake(() => {
+                        return Promise.resolve(new Response("{}", { status: 439, statusText: "Too Many Requests" }));
+                    });
+
+                    const config = this.createSenderConfig(TransportType.Fetch);
+                    const { sender } = this.initializeCoreAndSender(config, "000e0000-e000-0000-a000-000000000000");
+
+                    const telemetryItem: ITelemetryItem = {
+                        name: "fake item",
+                        iKey: "testIkey2;ingestionendpoint=testUrl1",
+                        baseType: "some type",
+                        baseData: {}
+                    };
+
+                    this.processTelemetryAndFlush(sender, telemetryItem);
+                }
+            ].concat(PollingAssert.createPollingAssert(() => {
+                if (this.statsbeatCountSpy.called && this.fetchStub.called) {
+                    this.assertStatsbeatCall(400, "failure");
+                    return true;
+                }
+                return false;
+            }, "Waiting for fetch sender and Statsbeat count to be called") as any)
+        });
+
+    
+
+        // this.testCaseAsync({
```

Comment:
As this is actually the "default" sender mechanism we really need to test this.

You will most likely need to use the fakeServer and the helper `this.sendJsonResponse(request, {}, 200)` to simulate the responses examples are in several tests.